risc0/zkvm/src/host/server/session.rs

```diff
@@ -80,6 +80,22 @@ pub struct Session {
 
     /// The system state of the final [MemoryImage] at the end of execution.
     pub post_state: SystemState,
+
+    /// Path of temporary segment storage used internally.
+    temp_segment_dir: Option<PathBuf>,
```

Comment:
I think what we discussed was to have something like `Rc<TempDir>` on the `FileSegmentRef`.